### PR TITLE
Add tracker sandbox

### PR DESCRIPTION
**Pull request by dan-jones-iw**

### CODE REVIEW

File: coordinator/src/routes/tracker.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  status: string
+}
+
+interface trackerResponse extends Partial<fhir.Task> {
+  lineItem: Array<lineItem>
+}
+
+const sandboxSuccessResponse = (prescriptionId: string): trackerResponse => ({
+  resourceType: "Task",
+  groupIdentifier: fhir.createIdentifier("system", prescriptionId),
```
Comment (mfjarvis) [inline coordinator/src/routes/tracker.ts:16]: System should be "https://fhir.nhs.uk/Id/prescription-order-number" for short form prescription ID

File: coordinator/src/routes/tracker.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import Hapi from "@hapi/hapi"
+import {fhir} from "@models"
+import {BASE_PATH, ContentTypes} from "./util"
+
+interface lineItem {
```
Comment (mfjarvis) [inline coordinator/src/routes/tracker.ts:5]: Remove? This isn't part of a FHIR Task. Line item statuses will be returned by the GET /MedicationRequest interaction.

File: coordinator/src/routes/tracker.ts [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+const sandboxErrorResponse: fhir.OperationOutcomeIssue = {
+  severity: "error",
+  code: fhir.IssueCodes.INVALID,
+  diagnostics: "Please query on prescription-id."
+}
+
+export default [{
+  method: "GET",
+  path: `${BASE_PATH}/$tracker`,
```
Comment (mfjarvis) [inline coordinator/src/routes/tracker.ts:42]: GET /Task?

File: coordinator/src/routes/tracker.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      id: "3",
+      status: "status"
+    }
+  ]
+})
+
+const sandboxErrorResponse: fhir.OperationOutcomeIssue = {
+  severity: "error",
+  code: fhir.IssueCodes.INVALID,
+  diagnostics: "Please query on prescription-id."
```
Comment (mfjarvis) [inline coordinator/src/routes/tracker.ts:37]: I'd avoid using "please" in an error message. Maybe say something like:
> The query parameter "prescription-id" is mandatory.

File: tests/e2e/pact/specs/sandbox/tracker.spec.ts [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+          state: "is not authenticated",
+          uponReceiving: "a valid FHIR message",
+          withRequest: {
+            headers: {
+              "Content-Type": "application/fhir+json; fhirVersion=4.0",
+              "X-Request-ID": requestId,
+              "X-Correlation-ID": correlationId
+            },
+            query: {
+              "prescription-id": testPrescriptionId
```
Comment (mfjarvis) [inline tests/e2e/pact/specs/sandbox/tracker.spec.ts:34]: Should this be `focus:identifier` or have I missed something?

File: tests/e2e/pact/specs/sandbox/tracker.spec.ts [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  /* eslint-disable  @typescript-eslint/no-explicit-any */
+  async (provider: any) => {
+    const client = () => {
+      const url = `${provider.mockService.baseUrl}`
+      return supertest(url)
+    }
+
+    describe("tracker e2e test", () => {
+      test("should return 200", async () => {
+        const apiPath = `${basePath}/$tracker`
```
Comment (mfjarvis) [inline tests/e2e/pact/specs/sandbox/tracker.spec.ts:18]: Should be /Task

File: tests/e2e/pact/specs/sandbox/tracker.spec.ts [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        }
+
+        await provider.addInteraction(interaction)
+        await client()
+          .get(apiPath)
+          .set("Content-Type", "application/fhir+json; fhirVersion=4.0")
+          .set("Accept", "application/fhir+json")
+          .set("X-Request-ID", requestId)
+          .set("X-Correlation-ID", correlationId)
+          .query({"prescription-id": testPrescriptionId})
```
Comment (mfjarvis) [inline tests/e2e/pact/specs/sandbox/tracker.spec.ts:54]: Same here

File: coordinator/src/routes/tracker/task.ts [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+    "Prescription"
+  )
+  const medicationIdentifier = fhir.createIdentifierReference(fhir.createIdentifier(
+    "https://tools.ietf.org/html/rfc4122",
+    "ee86a018-779c-4809-999f-a9d89cdfd30f"
+  ))
+  const medicationStatus: Array<fhir.ExtensionExtension<fhir.CodeableConceptExtension>> = [{
+    url: "https://fhir.nhs.uk/StructureDefinition/Extension-EPS-DispensingInformation",
+    extension: [{
+      url: "dispenseStatus",
```
Comment (mfjarvis) [inline coordinator/src/routes/tracker/task.ts:21]: Can't see this extension in the spec: https://simplifier.net/ukdigitalmedicine/extensionrepeatdispensinginformation

Maybe Kevin hasn't added it yet?

Comment (dan-jones-iw) [reply coordinator/src/routes/tracker/task.ts:21]: seems like this is the correct extension https://simplifier.net/nhsdigitalspine/extensionepsdispensinginformation

Comment (mfjarvis) [reply coordinator/src/routes/tracker/task.ts:21]: Ah I see, will ask Kevin to fix

**Review by mfjarvis — APPROVED** [review]

**Review by mfjarvis — APPROVED** [review]

**Pull request merged**